pkg/image/api/v1/types.go

```diff
@@ -144,3 +182,44 @@ type DockerImageReference struct {
 	Tag       string
 	ID        string
 }
+
+type ImageStreamImport struct {
+	unversioned.TypeMeta `json:",inline"`
+	kapi.ObjectMeta      `json:"metadata,omitempty" description:"metadata about the image stream, name is required"`
+
+	Repository *RepositoryImport `json:"repository,omitempty" description:"if specified, import a single Docker repository's tags to this image stream"`
+	Images     []ImageImport     `json:"images,omitempty" description:"a list of images to import into this image stream"`
+}
+
+type RepositoryImport struct {
+	From kapi.ObjectReference `json:"from" description:"the source for the image repository to import; only kind DockerImage and a name of a Docker image repository is allowed"`
+	To   bool                 `json:"to,omitempty" description:"if true, the located images should be added to the image stream with the corresponding tags"`
```

Comment:
Add details about what should happen when this is false.  I'm assuming that it would result in a `RepositoryImportStatus` that describes what would have been imported.